templates/democracy/test/democracy.js

```diff
@@ -1,402 +1,316 @@
+const Timestamp = artifacts.require('Timestamp')
 
-const getContract = name => artifacts.require(name)
 const pct16 = x => new web3.BigNumber(x).times(new web3.BigNumber(10).toPower(16))
-const getEventResult = (receipt, event, param) => receipt.logs.filter(l => l.event === event)[0].args[param]
-const getAppProxy = (receipt, id) => receipt.logs.filter(l => l.event === 'InstalledApp' && l.args.appId === id)[0].args.appProxy
-const getVoteId = (receipt) => {
-    const logs = receipt.receipt.logs.filter(l => l.topics[0] === web3.sha3('StartVote(uint256,address,string)'))
-    return web3.toDecimal(logs[0].topics[1])
-}
-
-contract('DemocracyTemplate', accounts => {
-    const ETH = '0x0'
-    let ens
-    let daoAddress, tokenAddress
-    let financeAddress, tokenManagerAddress, vaultAddress, votingAddress
-    let finance, tokenManager, vault, voting
-    let template, receiptInstance
-
-    const owner = accounts[0]
-    const holder20 = accounts[1]
-    const holder29 = accounts[2]
-    const holder51 = accounts[3]
-    const nonHolder = accounts[4]
-
-    const neededSupport = pct16(50)
-    const minimumAcceptanceQuorum = pct16(20)
-    const votingTime = 60
-
-    before('fetch democracy template', async () => {
-        if (await isLocalNetwork()) {
-            // transfer some ETH to other accounts
-            await web3.eth.sendTransaction({ from: owner, to: holder20, value: web3.toWei(10, 'ether') })
-            await web3.eth.sendTransaction({ from: owner, to: holder29, value: web3.toWei(10, 'ether') })
-            await web3.eth.sendTransaction({ from: owner, to: holder51, value: web3.toWei(10, 'ether') })
-            await web3.eth.sendTransaction({ from: owner, to: nonHolder, value: web3.toWei(10, 'ether') })
+const getVoteId = receipt => decodeEvents(receipt, Voting.abi, 'StartVote')[0].args.voteId
+const getAppProxy = (receipt, id) => decodeEvents(receipt, DemocracyTemplate.abi, 'InstalledApp').find(e => e.args.appId === id).args.appProxy
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+const ETH = ZERO_ADDRESS
+
+contract('Democracy', ([owner, holder20, holder29, holder51, nonHolder]) => {
+  let ens, token, dao, finance, tokenManager, vault, voting, template, receiptInstance, receiptToken
+
+  const TOKEN_NAME = 'DemocracyToken'
+  const TOKEN_SYMBOL = 'DTT'
+  const VOTING_TIME = 60
+  const REQUIRED_SUPPORT = pct16(50)
+  const ACCEPTANCE_QUORUM = pct16(20)
+  const STAKES = [20e18, 29e18, 51e18]
+  const HOLDERS = [holder20, holder29, holder51]
+
+  const increaseTime = async s => {
+    if (isGanache()) return timeTravel(s)
+    const previousTime = await (await Timestamp.new()).getNow()
+    await new Promise(resolve => setTimeout(resolve, s * 1000))
+    const currentTime = await (await Timestamp.new()).getNow()
+    assert.isAtLeast(currentTime.minus(s).toNumber(), previousTime.toNumber(), `sleep/time-travel helper failed to increase ${s} seconds`)
+  }
```

Comment:
Should we pull this off into a shared helper?

Reply:
sure! will do that 👍